site/cypress/e2e/cypress-feature/late_days_cache.spec.js

```diff
@@ -0,0 +1,462 @@
+        });
+
+        it('Changes gradeable due date back', () => {
+            cy.get('#date_due')
+                .clear()
+                .type('1972-01-01 03:59:59')
+                .click();
+            cy.get('#late_days').click(); // Dismiss calender and trigger save
+
+            cy.get('#save_status', {timeout:10000}).should('have.text', 'All Changes Saved');
+
+            cy.visit(['sample', 'bulk_late_days']);
+
+            cy.get('#late-day-table > tbody > tr > [data-before-content="Late Allowed Homework"] ~')
+                .then((cell) => expect(cell.text().trim()).to.equal(''));
+        });
+
+        it('Disables gradeable due date', () => {
+            cy.get('#has_due_date_no').check();
+
+            cy.get('#save_status', {timeout:10000}).should('have.text', 'All Changes Saved');
+
+            cy.visit(['sample', 'bulk_late_days']);
+
+            // Bulk late days should not have gradeable title
+            cy.get('#late-day-table > tbody > tr > [data-before-content="Late Allowed Homework"]').should('have.length', 0);
+
+        });
+
+        it('Re-enables gradeable due date', () => {
+            cy.get('#has_due_date_yes').check();
+
+            cy.get('#save_status', {timeout:10000}).should('have.text', 'All Changes Saved');
+
+            cy.visit(['sample', 'bulk_late_days']);
+
+            // Bulk late days should have gradeable title
+            cy.get('#late-day-table > tbody > tr > [data-before-content="Late Allowed Homework"]').should('have.length.gt', 0);
+
+        });
+
+        it('Disables late days', () => {
+            cy.get('#no_late_submission').check();
+
+            cy.get('#save_status', {timeout:10000}).should('have.text', 'All Changes Saved');
+
+            cy.visit(['sample', 'bulk_late_days']);
+
+            cy.get('#late-day-table > tbody > tr > [data-before-content="Late Allowed Homework"]').should('have.length', 0);
+
+        });
+
+        it('Re-enables late days', () => {
+            cy.get('#yes_late_submission').check();
+
+            cy.get('#save_status', {timeout:10000}).should('have.text', 'All Changes Saved');
+
+            cy.visit(['sample', 'bulk_late_days']);
+
+            cy.get('#late-day-table > tbody > tr > [data-before-content="Late Allowed Homework"]').should('have.length.gt', 0);
+
+        });
+
+        it('Changes late days allowed', () => {
+            cy.get('#late_days')
+                .clear()
+                .type('1')
+                .click();
+            cy.get('#date_due').click(); // Dismiss calender and trigger save
+
+            cy.get('#save_status', {timeout:10000}).should('have.text', 'All Changes Saved');
+
+            cy.visit(['sample', 'bulk_late_days']);
+
+            cy.get('#late-day-table > tbody > tr > [data-before-content="Late Allowed Homework"] ~')
+                .then((cell) => expect(cell.text().trim()).to.equal(''));
+        });
+    });
```

Comment:
These should be combined into a bigger test, or two tests, one for due date, one for late days